forest/sycamore/common.py

```diff
@@ -782,6 +791,82 @@ def read_aggregate_answers_stream(
         logger.warning("No survey_answers data found")
         return pd.DataFrame(columns=["Local time"], dtype="datetime64[ns]")
 
+    aggregated_data = fix_radio_answer_choices(aggregated_data, config_path,
+                                               history_path)
+
+    android_radio_questions = aggregated_data.loc[
+        # find radio button questions. These are the ones that show up weird
+        # on Android.
+        (aggregated_data["question type"] == "Radio Button Question")
+        # only include answers after they started listing radio button answers
+        # as ints in the Android app
+        & (aggregated_data["UTC time"] > ANDROID_NULLABLE_ANSWER_CHANGE_DATE)
+        # they will have ints in their answer field.
+        & (aggregated_data["answer"].apply(
+            lambda x: x.isdigit() if isinstance(x, str)
+            else True if isinstance(x, int)
+            else False)),
+        "answer"
+        # if their question type looks like this, it was made on an Android
+        # device. Radio button questions are the only ones with the integer
+        # instead of text (Also, avoiding any possible text outputs)
+    ].index
```

Comment:
This is a huge statement. It probably makes sense to simplify it.

Reply:
Hopefully what I did makes more sense...